PrestaShop/PrestaShop
# Introduce single unified UpdateProductCommand

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | related POC's #29999, #30009, #30025. Current approach having multiple CQRS commands for product update (commands like UpdateProductBasicInformationCommand, UpdateProductPricesCommand, UpdateProductOptionsCommand, UpdateProductDetailsCommand, UpdateProductSeoCommand) has some flaws - user saves product once but it is actually fetched and saved couple times and the product update hooks fires multiple times too. This PR introduces single command for all these general properties update (except for all the association commands like setting categories, suppliers, features etc.). Integrated 2 commands for now: **basic information** and **options**. The new product_commands_builder (which builds the new command) tags are commented out, so it doesn't affect the product page yet. This approach should eventually replace the currently existing commands, but for now we are still keeping the old multi-command approach working in product page until this new one is fully implemented.
| Type?             | refacto
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | 
| Related PRs       | 
| How to test?      | see bellow
| Possible impacts? | 

**How to test**
Locally it can be tested by uncomenting tags of product_commands_builder (in Resources/config/services/core/form/command_builder.yml **prestashop.core.form.command_builder.product.update_product_commands_builder**) and making sure that the UpdateProductCommand appears in the list of generated commands in **Core/Form/IdentifiableObject/DataHandler/ProductFormDataHandler::update()**. Also related behat scenarios can be run using these commands in terminal:
 - ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s update_product --tags update-basic-information
 - ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s update_product --tags update-multi-shop-basic-information
 - ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s update_product --tags update-options
 - ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s update_product --tags update-multi-shop-options
